### PR TITLE
Convert issue templates to forms

### PR DESCRIPTION
**Pull request by braem**
Converts our issue templates to issue forms (oooo shiny)

The only thing I don't like about them is that [you can't paste in images from the clipboard or embed them](https://github.com/github/feedback/discussions/5053), unless you edit them in post-creation.. that honestly might be a dealbreaker for now.

Those that make issues a lot (lookin at you @hexaflexahexagon) should review this. You can play around with them here: https://github.com/braem/issue-testing/issues

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/enhancement-request.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Enhancement Request
+description: Suggest an improvement to be made for an existing feature
+title: "[ENHANCEMENT]: "
+labels: ["Type: Enhancement"]
+body:
+- type: textarea
+  attributes:
+    label: What feature is your improvement idea related to?
+    description: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...].
```
Comment (hexaflexahexagon) [inline .github/ISSUE_TEMPLATE/enhancement-request.yml]: The `Ex. ...` section could turn into 
```
    placeholder: |
      I'm always frustrated when [...].
```
like on the bug template?

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Feature Request
+description: Suggest an idea for this project
+title: "[FEATURE]: "
+labels: ["Type: Feature"]
+body:
+- type: textarea
+  attributes:
+    label: Is your feature request related to a problem?
+    description: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...].
```
Comment (hexaflexahexagon) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: Same as above the `Ex. ...` section could turn into 
```
    placeholder: |
      I'm always frustrated when [...].
```
like on the bug template

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: Bug Report
+description: Create a report to help us improve the game
+title: "[BUG]: "
```
Comment (hexaflexahexagon) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: Do we want to prefix issues with [ASDF:]? None of our current issues follow that pattern and we already have cards for bug/enhancement/feature so it's not like we need that for filtering. Same goes for the other 3 templates

**Review by hexaflexahexagon — COMMENTED** [review]
not being able to upload images really sucks, i hope they add something for that. if you edit an issue you've already made it looks like it just gets converted into normal formatting where you're able to upload pictures all the same, just not within the code block. 
image
https://github.com/braem/issue-testing/issues/6

maybe instead we could just update our issue prefabs to look like this instead? sorta works, but you lose the pretty descriptions and clean boxes that the new template has.



### Describe the bug

```markdown

```


### Relevant Screenshots or Videos



### How To Reproduce

```markdown
1. 
2. 
3. 
```


### Expected Behavior

```markdown

```


### Operating System

```markdown

```

### Renderer [Pick one]

DX9 or DX11

### UI Framework [Pick one]

VGUI or Panorama

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 3 above)
Comment (braem) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: An annoying thing is that the title for the new forms cannot start empty.. very weird limitation

**Comment by braem** [comment]
Yeah no images are a dealbreaker imo. The feature is in beta after all, so safe to expect improvements relatively soon. I am hopeful:
image
(thinkin someone assigned this to themself internally?)

Don't think it's worth it updating the templates outside of this; thinking some people would be confused about the \`\`\`markdown stuff

**Comment by braem** [comment]
This in limbo now. Can close in hopes of the future or leave as draft. Dont delete branch though pls!

**Comment by hexaflexahexagon** [comment]
Agreed.

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 3 above)
Comment (braem) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: Actually it is optional, it just cannot be empty.

**Comment by braem** [comment]
If you don't include the `render` field you can insert images like normal, woops. See https://github.com/ChaosInitiative/Chaos-Source/pull/208

**Comment by hexaflexahexagon** [comment]
😳

will check in a few days

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      When I do X then Z should happen.
+  validations:
+    required: true
+    
+- type: textarea
+  attributes:
+    label: Screenshots
+    description: If applicable, add screenshots to help explain your problem.
+  validations:
+    required: false
```
Comment (hexaflexahexagon) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: a thought ive had a few times is if we should remove the "screenshot" section completely. at least for how i write issues i'll tend to write words, add an image, then write more words. leaving it blank will make something like this 

https://github.com/braem/issue-testing/issues/10

which is probably fine, just a bit ugly.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  validations:
+    required: true
+- type: dropdown
+  attributes:
+    label: UI Framework
+    options:
+      - VGUI
+      - Panorama
+  validations:
+    required: true
```
Comment (hexaflexahexagon) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: also maybe it'd be good to add a small explanation of how to tell which of these options you're using? i think some people get confused which is why they leave it blank, though this menu helped that already a good bit so i'm not 100% on it

image

**Review by hexaflexahexagon — COMMENTED** [review]
looks good to me now that you can add pictures. left a few optional ideas to consider but seems nice as it is anyways

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 5 above)
Comment (braem) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: maybe like `VGUI (default)`?

Comment (hexaflexahexagon) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: 👍

**Review by hexaflexahexagon — APPROVED** [review]

**Review by Gocnak — APPROVED** [review]

**Pull request merged**